packages/cli/src/commands/db.ts

```diff
@@ -57,6 +61,66 @@ export const runMigrate = async () => {
   })
 }
 
+export async function resetPostgres(connectionString: string): Promise<void> {
+  const dbName: string = getDbName(connectionString)
+  try {
+    // close all other connections
+    await db.raw(
+      `SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE pid <> pg_backend_pid() AND datname='${dbName};'`,
+    )
+    // currently assuming the public schema is being used
+    // delete schema and recreate with the appropriate privileges
+    await db.raw('DROP SCHEMA public cascade;')
+    await db.raw('CREATE SCHEMA public;')
+    await db.raw('GRANT ALL ON schema public TO postgres;')
+    await db.raw('GRANT ALL ON schema public TO public;')
+    // run migration
+    await runMigrate()
+    console.log('Your database has been reset.')
```

Comment:
This all looks great! Love the clarity with the platform-specific implementations. As a note to myself, we'll need to eventually support customizing this hook in database plugins in the future.

Would you be able to use the built-in logging utilities we have in `@blitzjs/server/utils/log.ts` rather than `console.log`/`console.error`? If everything is passing through those log utilities we can control the look and feel of the CLI a bit better

Reply:
Wasn't sure what to do with logging so thanks for the heads up. I'm currently just logging the errors when the reset fails but should I replace them by more generic messages like "Database reset failed" or something? The database errors are pretty clear about what's going wrong but maybe it's too much clutter in the CLI?

Reply:
@LoriKarikari maybe both? Let's try something like this:
```
log.error('[[insert operation name here]] failed with an error from the database:')
log.error(error)
```
we may need to update the type of `log.error` to handle `Error` objects, I think right now it only accepts `string`, but the console knows how to stringify `Error`s so we should be able to pass those as well